### DISCUSSION TITLE
Is it possible to use the DateRangePicker without the Button?

### DISCUSSION THREAD

**Discussion by ChristophP** [adobe/react-spectrum#4918] [Q&A]
Hi,

thanks for making this great library. Really helps to not have to worry about all the ARIA patterns yourself.

We're in need of a DateRangePicker. We would like our picker to open when the Input is clicked rather than having a button next to it, similar to how the MUI DateRangePicker behaves (here's an [example](https://mui.com/x/react-date-pickers/date-range-picker/)) .

So starting from this...

![image](https://github.com/adobe/react-spectrum/assets/5880236/db7293d1-ad07-4bfd-8a32-8c08e0b5cb82)

... I tried briefly to make this happen by removing the `Button` and using click handlers to control on the `DateInput`s the `isOpen` prop  without success. I then resorted to using `onFocus` events on the `DateInput`s which worked for opening but not to open it a second time since the focus on the input stays after closing.
Now I am unsure if I'm trying to do something the DateRangePicker was not meant for.

Would appreciate some pointers on how to best achieve this or if this is something to stay away from.

**Comment by snowystinger** [comment]
You could make the entire field a button with the formatted date range and just open to a Range Calendar
Something along these lines https://codesandbox.io/s/hungry-marco-fmgv9p?file=/src/index.js

If you're really trying to keep the input, you may want to refer to this issue for some troubles you might run into or just things to consider https://github.com/adobe/react-spectrum/issues/4723

**Reply by snowystinger** [reply]
Similar discussion here https://github.com/adobe/react-spectrum/discussions/4914#discussioncomment-6735371
